src/rust/lib/graph-editor/src/component/operator.rs

```diff
@@ -0,0 +1,271 @@
+//! Provides generic operations that can be applied to UI components.
+//!
+//! Rationale: this is a step towards a higher level abstraction for arbitrary UI elements. That is,
+//! instead of doing a lot of low level functionality within our UI components, we should instead be
+//! able to access some data about them and manipulate them in a consistent manner. For example:
+//! instead of every component implementing a "fullscreen mode", they just provide resizing
+//! functionality and access to their su-components and instead of determining the layer of shapes
+//! within the UI component, we instead make the shapes accessible via an API that indicates the
+//! expected behaviour of the shapes, and the actual layer management happens outside of the
+//! component.
+//!
+//! This is also a step to avoid relying too much on the presence of the `Scene` within the UI
+//! component.
+use crate::prelude::*;
+
+use ensogl::display::DomScene;
+use ensogl::display::DomSymbol;
+use ensogl::display::Scene;
+use ensogl::display::Symbol;
+use ensogl::display::scene::View;
+use ensogl::display::traits::*;
+use ensogl::display;
+use ensogl::frp;
+use ensogl::gui::component::animation;
+
+
+
+// ==================================
+// === UI Component Helper Traits ===
+// ==================================
+
+/// Should be implemented by UI component that consist of `Symbol`. Provides access to the shapes
+/// and some helper methods for working with those shapes.
+pub trait NativeComponent {
```

Comment:
After readign the code for 1 hour I un derstood that the logic for `NativeComponent` is in fact to manage its z-index only. Please rename it accordingly to this knowledge. hen it ill grow bigger and will have more responsibilities, we can rename it again. No its only for z-indexing.

Reply:
Moreover, until we got proper z-indexing management, I would move it for now to `component::visualization::` space

Reply:
Actually the logic is even moe complex here. `NativeComponent` is both something that provides you with symbols and also layouts them. I would extract `HasSymbols` as separate trait and combine it with other traits to get the desired effect.